lib/vector-config/src/schema.rs

```diff
@@ -232,12 +231,12 @@ where
     })
 }
 
-pub fn generate_set_schema<T>(gen: &RefCell<SchemaGenerator>) -> Result<SchemaObject, GenerateError>
-where
-    T: Configurable,
-{
-    // Generate the actual schema for the element type `T`.
-    let element_schema = get_or_generate_schema::<T>(gen, None)?;
+pub(crate) fn generate_set_schema(
+    config: &ConfigurableRef,
+    gen: &RefCell<SchemaGenerator>,
+) -> Result<SchemaObject, GenerateError> {
+    // Generate the actual schema for the element type.
+    let element_schema = get_or_generate_schema(config, gen, None)?;
```

Comment:
Why are we making this function, and some others, only public in the crate itself? Clippy lints?

Reply:
No particular reason, other than going through and seeing which are actually used outside of the crate.

Reply:
Works for me. 🤷🏻